ZEN.md

```diff
@@ -0,0 +1,65 @@
+# Mobile RUM SDK Philosophy
+
+## TL;DR;
+
+> Always keep in mind that this library lives on our users' users' device, within our user's applications. We need to be **stable**, **efficient** and **transparent**.
+
+## Foreword
+
+This SDK lives in our customer’s applications, and is run on end users devices. Because those devices can range from low to high end, and can be used in varying conditions of network and battery, we need to make sure our footprint on the end user experience is as small as possible.
+
+## Small Footprint
+
+- **Runtime performance**
+    - Do not perform unnecessary operation;
+    - Every public operation must be executed fast;
+    - Delegate heavy work to background threads/workers when possible;
+- **Library size**
+    - Avoid unnecessary dependencies;
+- **Network load**
+    - Batch requests as much as possible;
+    - Avoid making request when the device is asleep or has low battery;
+
+## Stability
+
+- **Zero crash caused by our code!**
+    - Unless the crash is on top-level method, caused by identifiable developer mistake, and notified with an understandable error message;
+    - Even in top-level methods, prefer making the library non operating and logging the issue in the console logs, rather than throwing an exception
```

Comment:
💅 
```suggestion
    - Unless the crash is on top-level method, caused by identifiable developer mistake, and notified with an understandable error message.
    - Even in top-level methods, prefer making the library non operating and logging the issue in the console logs, rather than throwing an exception.
```